[gmengqi] iP

gmengqi opened the pull request:
### Duke

> “Your mind is for having ideas, not holding them.” – David Allen [[source](https://dansilvestre.com/productivity-quotes)]

Duke frees your mind of having to remember things you need to do. It's,

1. text-based
2. user-friendly 
3. ~FAST~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x]  Managing tasks
- [x]  Managing deadlines 
- [x] Find tasks easily with keywords
- [ ]  Reminders (coming soon)

`public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
`

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

gmengqi committed:
greet, echo and exit

gmengqi committed:
add and list function implemented

gmengqi committed:
able to mark and unmark task

gmengqi committed:
added type of tasks

gmengqi committed:
semi-automated testing of Duke

gmengqi committed:
catch exceptions

gmengqi committed:
implemented delete command

gmengqi committed:
save task list in hard disk automatically

gmengqi committed:
change deadline dates as LOCALDATETIME

gmengqi committed:
fix merge conflict

gmengqi committed:
use more OOP

gmengqi committed:
organise into packages

gmengqi committed:
fix datetime issue in deadline

gmengqi committed:
set up gradle

gmengqi committed:
add Junit Tests for Parser and TaskList class

gmengqi committed:
create JAR file

gmengqi committed:
fornatted the code

gmengqi committed:
formatted the code

gmengqi committed:
add JavaDoc

gmengqi committed:
edit code to fit coding standard

gmengqi committed:
find tasks by searching keyword

gmengqi committed:
merge branch-Level-9

gmengqi committed:
merge branch-JavaDoc

gmengqi committed:
merge branch-A-CodingStandard

gmengqi committed:
use checkstyle to detect coding style violations

chia-yh commented on src/main/java/duke/task/Parser.java:
```diff
@@ -0,0 +1,110 @@
+
+public class Parser {
+    private static String[] arr;
+    private static final String TODO = "todo";
```
I like that 'magic' Strings are avoided here, as well as in other areas of your code.

chia-yh commented on src/main/java/duke/task/Parser.java:
```diff
@@ -0,0 +1,110 @@
+        String[] arr = splitCommand(command);
+        String commandType = arr[0];
+
+        switch (commandType) {
```
I like that arrowhead style code is avoided as much as possible.

chia-yh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,66 @@
+     * Start the Duke program.
+     */
+    public void run() {
+        ui.showWelcome();
```
I like that the code is structured logically.

chia-yh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,66 @@
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
```
I like the use of SLAP here.

chia-yh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,66 @@
+     * @throws DukeException
+     */
+    public static void main(String[] args) throws DukeException {
+        new Duke("ip/data/tasks.txt").run();
```
Perhaps turn ```ip/data/tasks.txt``` into a named constant, to avoid a "magic" String.

chia-yh commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,138 @@
+     */
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
```
Not sure I like that the method is rather long, perhaps find a way to abstract some of the execution?

chia-yh reviewed (commented):
Looks to be following the code quality guidelines well, with some minor potential issues, keep up the good work!

gmengqi committed:
fix storage issue where deleted task not updated in txt file

gmengqi committed:
fix minor bug

gmengqi committed:
implement GUI

gmengqi committed:
checkstyle with gradle

AngJunYang commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,61 @@
+     * @return String informing users on whether the task is completed.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```
I like the use of comments to improve readability

AngJunYang commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,138 @@
+ */
+
+public class Ui {
+    private static String logo =
+            "      _     _            _                 \n" +
+                    "     | |   (_)          | |                \n" +
+                    "  ___| |__  _ _ __   ___| |__   __ _ _ __  \n" +
+                    " / __| '_ \\| | '_ \\ / __| '_ \\ / _` | '_ \\ \n" +
+                    " \\__ \\ | | | | | | | (__| | | | (_| | | | |\n" +
+                    " |___/_| |_|_|_| |_|\\___|_| |_|\\__,_|_| |_|\n";
```
I like the logo design!

AngJunYang commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,178 @@
+        this.filePath = filePath;
+    }
+
+    /**
+     * Convert the task information in the txt file to Task objects.
+     *
+     * @param str
+     * @return Task objects.
+     */
```
I like the use of Javadoc comments even for private functions

AngJunYang commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,69 @@
+     * @param num
+     * @return the Task object.
+     */
+
```
Perhaps remove the blank line between documentation block and method/class?

AngJunYang reviewed (commented):
I love the icons you have used, they are very cute! Code looks good with minimal coding errors!

gmengqi committed:
generate new JAR file

gmengqi committed:
use assertions to document important assumptions

gmengqi committed:
enable assertion in intelliJ

gmengqi committed:
Add command and Parser: make the code adhere to the code standard and quality.

Add command class and Parser class comprise of codes that are similar to each other and have methods that are too long such as execute and convertToStr.

By refactoring the code and extracting the parts where the codes repeat itself, it makes the code more readable.

Let's,
* refactor the code in the execute method in AddCommand class using extract method.
* refactor the code in convertToStr method in Parser class.

gmengqi committed:
Merge pull request #1 from gmengqi/branch-A-Assertions

Add assertions

gmengqi committed:
merge master into this branch

gmengqi committed:
Merge pull request #2 from gmengqi/branch-A-CodeQuality

Code quality

gmengqi committed:
The users unable to receive reminders when they are upcoming deadlines.

Due to the lack of reminders, some forgetful users might miss their deadlines.

Lets:
Create a reminder feature to remind users about any upcoming reminders that are less than 24 hours away from the deadline so the users have sufficient time to do the task.

gmengqi committed:
Merge branch 'branch-B-Reminders' into master

gmengqi committed:
able to edit the tasks

gmengqi committed:
update main and mainwindow to show reminder also

gmengqi committed:
add help command

gmengqi committed:
Merge branch 'branch-C-Help' into master.

gmengqi committed:
edit task list to make it neater

gmengqi committed:
fix reminder feature bug

gmengqi committed:
merge branch-B-Reminders into master

gmengqi committed:
edit README.md

gmengqi committed:
customise shinchan

gmengqi committed:
code checkstyle

gmengqi committed:
fix all bugs found

gmengqi committed:
improve GUI

gmengqi committed:
update Ui.png

gmengqi committed:
fix more bugs found and improve GUI

gmengqi committed:
edit the javadocs to coding standard

gmengqi committed:
edit README.md to correct markdown format

gmengqi committed:
Update README.md

gmengqi committed:
fix more bugs found

gmengqi committed:
Merge branch 'master' of github.com:gmengqi/ip

gmengqi committed:
fix bug: not showing entire list for GUI
